platform/ABProcess.js

```diff
@@ -93,6 +95,35 @@ module.exports = class ABProcess extends ABProcessCore {
       return this.instanceUpdate(instance);
    }
 
+   /**
+    * Save a copy of the definitions in the DB. If the definition already exists
+    * find the uuid
+    * @returns {Object} the ProcessDefinition
+    */
+   async instanceDefinition(req) {
+      try {
+         const definition = convert.xml2js(this.xmlDefinition, {
+            compact: true,
+         });
+         const defHash = hash(definition);
+         const processDefModel = this.AB.objectProcessDefinition().model();
+         let [defRecord] = await processDefModel.find({ hash: defHash }, req);
+         if (!defRecord) {
+            defRecord = await processDefModel.create({
+               hash: defHash,
+               definition,
+            });
+         }
+         return defRecord;
+      } catch (error) {
+         this.AB.notify.developer(error, {
+            context: "ABProcess.instanceDefinition",
+            process: this,
```

Comment:
minor issue: but now when I go through error dumps, I prefer to reduce the amount of info being dumped.  So a `this.toObj()` would be better here than the raw object and all the additional data it includes.